File: JavaNgArch/src/javangarch/domain/domainmodel/gn/Func.java
package javangarch.domain.domainmodel.gn;

import java.util.Date;

import lombok.Getter;
import lombok.Setter;
import projectbase.domain.BaseDomainObject;
import projectbase.sharparch.domain.domainmodel.DomainSignature;
@Getter@Setter
public class Func extends BaseDomainObject {
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	public static final String DefaultSort = "Name asc";

	private String name;
	@DomainSignature
	private String level;
	private String code;

}

File: NetNgArch/src/NetNgArch.Domain/DomainModel/GN/Func.cs
using System;
using System.Collections.Generic;
using ProjectBase.Domain;
using ProjectBase.Utils;
using SharpArch.Domain.DomainModel;

namespace NetNgArch.Domain.DomainModel.GN
{
    public class Func : BaseDomainObject
    {
        public static readonly SortStruc<Func>[] DefaultSort = new SortStruc<Func>[]
                                                                    {
                                                                        new SortStruc<Func>{
                                                                            OrderByExpression=o => o.Name,
                                                                            OrderByDirection = OrderByDirectionEnum.Asc
                                                                        }
                                                               };
        #region "persistent properties"

		public virtual string Name { get; set; }
		[DomainSignature]
		public virtual string Level { get; set; }
		public virtual string Code { get; set; }

        public virtual IList<Corp> Corps { get; set; }
        public virtual IList<Dept> Depts { get; set; }
        public virtual IList<User> Users { get; set; } 
     
        #endregion

        #region "methods"

        #endregion

    }

}

